staging/src/k8s.io/apiserver/pkg/endpoints/filters/limit_request.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"net/http"
+)
+
+// WithLimitUntilHealthy bloxks all requests to the handler unless the API is healthy for the first time or it is a health or liveness check
+func WithLimitUntilHealthy(handler http.Handler, c *rest.Config) http.Handler {
+	healthyFirstTime := false
+	coreClient, err := corev1client.NewForConfig(c)
+	if err != nil {
+		klog.Info("Unable to create client: &q", err)
+		return handler
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		allowedURLS := map[string]struct{}{
+			"/healthz": {},
+			"/livez":   {},
+			"/readyz":  {},
+		}
+
+		if _, ok := allowedURLS[r.URL.Path]; ok {
+			handler.ServeHTTP(w, r)
+			return
+		}
+
+		if !healthyFirstTime {
+			var code int
+			coreClient.RESTClient().Get().AbsPath("/healthz").Do(context.TODO()).StatusCode(&code)
+			if code != http.StatusInternalServerError {
```

Comment:
This doesn't seem right... can't healthz return something other than `http.StatusInternalServerError` and a 200?

Reply:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go#L171

This seems to be the handler for the health route.

Reply:
`/healthz`  will return 429  if reach maxinflight